ESPHomeRoombaComponent.h

```diff
@@ -99,9 +101,11 @@ class RoombaComponent : public PollingComponent, public CustomAPIDevice {
           Roomba::SensorVoltage,        // 2 bytes, mV, unsigned
           Roomba::SensorCurrent,        // 2 bytes, mA, signed
           Roomba::SensorBatteryCharge,  // 2 bytes, mAh, unsigned
-          Roomba::SensorBatteryCapacity // 2 bytes, mAh, unsigned
+          Roomba::SensorBatteryCapacity, // 2 bytes, mAh, unsigned
+          Roomba::SensorBatteryTemperature // 2 bytes, °C, signed
+
       };
-      uint8_t values[] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+      uint8_t values[] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
```

Comment:
shouldn't we need more room for _2 bytes signed_?

Reply:
Do not really know. It works like this.

Reply:
To my understanding here we allocate memory to receive values read for sensors.
So we should pay attention to give the right amount of bytes (zeroes) based on the depth we declared for every sensor:

![image](https://user-images.githubusercontent.com/447917/151715837-242f0e2a-4aba-43bd-90c0-1e1167d107e7.png)

i.e. please note how in this screenshot the _charging_ sensor is the only one using an int8 value (vs int16), so that it takes only one 0 into the buffer (vs two 0 for other sensors, hence leading to 11 _zeroes_).

Note also how the value for 2 bytes sensors comes with a sum of two values (plus a shift obtained adding 256). I suspect if we don't take care of this, we could run into issues as soon as we add another sensor, or even if the temperature value doesn't fit into a single byte

Reply:
Oh, sorry, I've just checked the [open interface specs](https://www.irobotweb.com/-/media/MainSite/Files/About/STEM/Create/2018-07-19_iRobot_Roomba_600_Open_Interface_Spec.pdf): they give 1 byte for temperature
This would explain why you get the right value even with a single byte